highlightjs/highlightjs.php

```diff
@@ -0,0 +1,51 @@
+<?php
+/**
+ * Name: Syntax Highlighting
+ * Description: Highlights syntax of code blocks with highlight.js
+ * Version: 1.0
+ * Author: Hypolite Petovan <hypolite@mrpetovan.com>
+ */
+
+use Friendica\App;
+use Friendica\Core\Addon;
+
+function highlightjs_install()
+{
+	Addon::registerHook('page_end', __FILE__, 'highlightjs_page_end');
+}
+
+function highlightjs_uninstall()
+{
+	Addon::unregisterHook('page_end', __FILE__, 'highlightjs_page_end');
+}
+
+function highlightjs_page_end(App $a, &$b)
+{
+	$basedir = $a->get_baseurl() . '/addon/highlightjs/asset';
+
+	if ($a->getCurrentTheme() == 'frio') {
+		$style = 'bootstrap';
+	} else {
+		$style = 'default';
+	}
+
+	$a->page['htmlhead'] .= <<< HTML
+
+<link rel="stylesheet" href="{$basedir}/styles/{$style}.css">
+
+HTML;
+
+	$b .= <<< HTML
+
+<script type="text/javascript" src="{$basedir}/highlight.pack.js"></script>
+<script type="text/javascript">
```

Comment:
I would prefer to put the following javascript code into an own js file. We should avoid to mix php and js if it is possible

Reply:
They are cached, but the first load is more expensive when it doesn't necessarily has to be.

I'll see what I can do.